File: algorithms/[448]找到所有数组中消失的数字/solution.py
# topics = ["数组"]

from typing import List


class Solution:
    def findDisappearedNumbers(self, nums: List[int]) -> List[int]:
        # 原地修改
        for num in nums:
            if nums[abs(num) - 1] > 0:
                nums[abs(num) - 1] *= -1

        res = []

        for i in range(len(nums)):
            if nums[i] > 0:
                res.append(i + 1)

        return res

File: algorithms/[448]找到所有数组中消失的数字/solution.ts
// topics = ["数组"]

function findDisappearedNumbers(nums: number[]): number[] {
  const res = Array.from({ length: nums.length }, (_, i) => i + 1);

  for (const num of nums) {
    res[num - 1] = 0;
  }

  return res.filter((i) => i != 0);
}
